non-oss/pharos_pro/license_key_command_option.rb

```diff
@@ -56,17 +56,26 @@ def cluster_info_configmap
           signal_error "Failed to get cluster-info configmap: #{ex.message}"
         end
 
+        def pharos_config_configmap
+          Dir.chdir(config_yaml.dirname) do
+            master_host.transport.connect
+            master_host.transport.exec!('kubectl get configmap --namespace kube-system -o yaml pharos-config')
+          end
+        rescue Pharos::ExecError => ex
+          signal_error "Failed to get cluster-info configmap: #{ex.message}"
+        end
+
         def cluster_id
           cluster_info.dig('metadata', 'uid')
         end
 
         def cluster_name
           Pharos::YamlFile.new(
-            StringIO.new(cluster_info.dig('data', 'kubeconfig')),
-            override_filename: "#{master_host.address}:kube-public/cluster-info.data.kubeconfig"
-          ).load.dig('clusters', 0, 'name')
+            StringIO.new(pharos_config_configmap),
```

Comment:
Hmm. Can't we just use `load_config.name` now as we have #1134 ?

Reply:
it would require that name is present in `cluster.yml` or user must pass it via options. I think it's more straightforward to read it from the cluster.

Reply:
Actually, the cluster name and the configmap would both be there if we run GatherFacts like we do in `pharos reset`:

```ruby
    def cluster_manager
      @cluster_manager ||= ClusterManager.new(load_config).tap do |cluster_manager|
        puts "==> Sharpening tools ...".green
        cluster_manager.load
        cluster_manager.gather_facts
      end
    end
```

After this, `cluster_manager.context['previous-config-map']` should contain the current `pharos-config` configmap and `load_config.name` will have the name from the cluster if it was not given.

Reply:
It will do some unnecessary extra stuff that is only required for Up, but it should also set the `config.master_host` to one that should be the most suitable.

Actually, maybe it should do:

```ruby
      @cluster_manager ||= ClusterManager.new(load_config.tap { |c| c.hosts.keep_if?(&:master)}).tap do |cluster_manager|
```

to only operate on masters.